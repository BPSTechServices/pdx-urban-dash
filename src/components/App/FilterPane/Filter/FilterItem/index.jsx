import React, { Component } from "react";
import PropTypes from "prop-types";
import { Toast, ToastHeader } from "reactstrap";

export default class FilterItem extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    selected: PropTypes.bool,
    callback: PropTypes.func
  };

  constructor(props) {
    super(props);

    this.activate = this.activate.bind(this);
    this.deactivate = this.deactivate.bind(this);

    this.title = "";
    this.category = "";
    this.selected = false;
    this.callback = t => console.log("FilterItem uninitialized callback: " + t);

    this.state = {
      activation: this.props.selected ? this.deactivate : this.activate
    };
  }

  activate() {
    this.props.callback({
      title: this.props.title,
      category: this.props.category
    });

    this.setState({
      activation: this.deactivate
    });
    return false;
  }

  deactivate() {
    this.props.callback({
      title: this.props.title,
      category: this.props.category
    });

    this.setState({
      activation: this.activate
    });
    return false;
  }

  render() {

    if (this.props.selected) {
      return (
        <Toast
          fade={false}
          onClick={this.state.activation}
          style={
            this.props.selected ? { background: "green" } : { background: "" }
          }
        >
          <ToastHeader
            toggle={this.props.callback}
            style={{ height: "2rem" }}
          >
            {this.props.title}
          </ToastHeader>
        </Toast>
      );
    }
    else {
      return (
        <Toast
          fade={false}
          onClick={this.state.activation}
          style={
            this.props.selected ? { background: "green" } : { background: "" }
          }
        >
          <ToastHeader
            style={{ height: "2rem" }}
          >
            {this.props.title}
          </ToastHeader>
        </Toast>
      );
    }
  }
}