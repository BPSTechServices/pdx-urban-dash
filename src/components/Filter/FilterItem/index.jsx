import React, { Component } from "react";
import PropTypes from "prop-types";
import { Toast, ToastHeader } from "reactstrap";

export default class FilterItem extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    selected: PropTypes.bool,
    callback: PropTypes.func
  };

  constructor(props) {
    super(props);

    this.activate = this.activate.bind(this);
    this.deactivate = this.deactivate.bind(this);

    this.title = "";
    this.category = "";
    this.selected = false;
    this.callback = t => console.log("FilterItem uninitialized callback: " + t);

    this.state = {
      activation: this.props.selected ? this.deactivate : this.activate
    };
  }

  activate() {
    this.props.callback({
      title: this.props.title,
      category: this.props.category
    });

    this.setState({
      activation: this.deactivate
    });
    return false;
  }

  deactivate() {
    this.props.callback({
      title: this.props.title,
      category: this.props.category
    });

    this.setState({
      activation: this.activate
    });
    return false;
  }

  render() {
    return (
      <Toast
        fade={false}
        onClick={this.state.activation}
        style={
          this.props.selected ? { background: "green" } : { background: "" }
        }
      >
        <ToastHeader
          toggle={!this.props.selected ? false : this.props.callback}
          style={{ height: "2rem" }}
        >
          {this.props.title}
        </ToastHeader>
      </Toast>
    );
  }
}