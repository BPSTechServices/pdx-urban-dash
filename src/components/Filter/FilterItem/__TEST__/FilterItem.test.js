import React from 'react';
import { mount, shallow } from 'enzyme';
import FilterItem from '../index';
import PropTypes from "prop-types";

const testFunc = () => { console.log(); };

describe('FilterItem', () => {
  test('render unselected', () => {
    const component = shallow(<FilterItem title={'testT'} category={'testC'}/>);
    expect(component).toMatchSnapshot();
  });

  test('render selected', () => {
    const component = shallow(<FilterItem title={'testT'} category={'testC'} selected={true}/>);
    expect(component).toMatchSnapshot();
  });

  it('Simulate lick', () => {
    const mockCallBack = jest.fn();
    const item = mount(<FilterItem title={'testT'} category={'testC'} callback={mockCallBack}/>)
    item.find(FilterItem).simulate('click');
    expect(mockCallBack).toHaveBeenCalledWith({title: 'testT', category: 'testC'});
  });

  test('activate', () => {
    const mockCallBack = jest.fn();
    const component = shallow(<FilterItem title={'testT'} category={'testC'} callback={mockCallBack}/>)
    const instance = component.instance();

    const result = instance.activate()
    
    expect(result).toBe(false);
    expect(mockCallBack).toHaveBeenCalledWith({title: 'testT', category: 'testC'});
    expect(component.state('activation')).toBe(instance.deactivate);
  });
  
  test('deactivate', () => {
    const mockCallBack = jest.fn();
    const component = shallow(<FilterItem title={'testT'} category={'testC'} callback={mockCallBack}/>)
    const instance = component.instance();
    
    const result = instance.deactivate()
    
    expect(result).toBe(false);
    expect(mockCallBack).toHaveBeenCalledWith({title: 'testT', category: 'testC'});
    expect(component.state('activation')).toBe(instance.activate);
  });
});