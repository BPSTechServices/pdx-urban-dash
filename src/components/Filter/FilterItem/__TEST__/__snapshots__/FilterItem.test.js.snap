// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem render selected 1`] = `ShallowWrapper {}`;

exports[`FilterItem render unselected 1`] = `ShallowWrapper {}`;
